### PR TITLE
Documenting Screwdriver's overall Architecture

### PR DESCRIPTION
**Pull request by stjohnjohnson**
screen shot 2016-06-27 at 4 48 59 pm
screen shot 2016-06-27 at 4 49 05 pm

### CODE REVIEW

File: docs/architecture/overall.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+![Workflow](assets/workflow.png)
+
+## Workflow
+
+1. **Commit new code**
+
+    User starts a new build by one of the following operations:
+
+    - User pushes code to GitHub on the monitored branch
```
Comment (d2lam) [inline docs/architecture/overall.md]: What's **monitored** branch?

File: docs/architecture/overall.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    - User pushes code to GitHub on an open pull request to the monitored
+    branch
+    - User opens a new pull request on GitHub to the monitored branch
```
Comment (tkyi) [inline docs/architecture/overall.md]: I would switch the order between this one and the one before it; seems like a more logical progression to me.
Also, would the pull request be **to** the monitored branch or **against** it?

File: docs/architecture/overall.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    - User tells Screwdriver (via API or UI) to rebuild a given commit
+
+2. **Notify Screwdriver**
+
+    Signed [webhooks](https://developer.github.com/webhooks/) notify
+    Screwdriver's API about the change.
+
+3. **Trigger execution engine**
+
+    Screwdriver starts a job on the specified execution engine passing the
```
Comment (tkyi) [inline docs/architecture/overall.md]: Can you give a couple examples of `execution engines` here

File: docs/architecture/overall.md (hunk 1 above)
Comment (stjohnjohnson) [inline docs/architecture/overall.md]: How should I better explain that?

Comment (nkatzman) [inline docs/architecture/overall.md]: Maybe add a part about creating a screwdriver project. What's required, and what it means with github and describe the "monitored branch".

And then in this commit code, reference that

**Comment by nkatzman** [comment]
👍  looks good

**Pull request merged**